### PR TITLE
Release Zebra to 1.0.0-beta.13

### PR DESCRIPTION
**Pull request by oxarbitrage**
Depends-On: #4830
Depends-On: #4828
Depends-On: #4840

### Which Crates to Increment

To check if any of the top-level crates need version increments:
1. Go to the zebra GitHub code page: https://github.com/ZcashFoundation/zebra
2. Check if the last commit was a Zebra version bump, or something else

<details>
   
Alternatively you can:
- Use the github compare tool and check the `main` branch against the last tag ([Example](https://github.com/ZcashFoundation/zebra/compare/v1.0.0-alpha.15...main))
- Use `git diff --stat <previous-release-tag> origin/main`

</details>
   
Once you know which crates have changed:
- [x] Increment the crates that have new commits since the last version update
- [x] Increment any crates that depend on crates that have changed
- [x] Keep a list of the crates that haven't been incremented, to include in the PR

### How to Increment Versions

Zebra follows [semantic versioning](https://semver.org).

Semantic versions look like: MAJOR`.`MINOR`.`PATCH[`-`TAG`.`PRE-RELEASE]
   
<details>

#### Pre-Release Crates

Pre-Release versions have a `TAG` like "alpha" or "beta". For example: `1.0.0-alpha.0`

1. Increment the `PRE-RELEASE` version for the crate.

#### Unstable Crates

Unstable versions have a `MAJOR` version of zero. For example: `0.1.0`

1. Follow stable crate versioning, but increment the `MINOR` version for breaking changes

#### Stable Crates

For example: `1.0.0`

Increment the first version component in this list, and reset the other components to zero:
1. MAJOR versions for breaking public API changes and removals
    * check for types from dependencies that appear in the public API
2. MINOR versions for new features
3. PATCH versions for bug fixes
    * includes dependency updates that don't impact the public API

### Reviewing Version Bumps

Check for missed changes by going to:
`https://github.com/ZcashFoundation/zebra/tree/<commit-hash>/`
Where `<commit-hash>` is the hash of the last commit in the version bump PR.

If any Zebra or Tower crates have commit messages that are **not** a version bump, we have missed an update.
Also check for crates that depend on crates that have changed. They should get a version bump as well.

</details>
   
### Version Locations

Once you know which versions you want to increment, you can find them in the:
- [x] zebra* `Cargo.toml`s
- [x] tower-* `Cargo.toml`s
- [x] `zebra-network` protocol user agent: https://github.com/ZcashFoundation/zebra/blob/main/zebra-network/src/constants.rs
- [x] `README.md`
- [x] `book/src/user/install.md`
- [x] `Cargo.lock`: automatically generated by `cargo build`

#### Version Tooling

You can use `fastmod` to interactively find and replace versions.

For example, you can do something like:
```
fastmod --extensions rs,toml,md --fixed-strings '1.0.0-beta.11' '1.0.0-beta.12'
fastmod --extensions rs,toml,md --fixed-strings '0.2.26' '0.2.27' tower-batch tower-fallback
```

If you use `fastmod`, don't update versions in `CHANGELOG.md`.

## README

We should update the README to:
- [x] Remove any "Known Issues" that have been fixed
- [x] Update the "Build and Run Instructions" with any new dependencies.
      Check for changes in the `Dockerfile` since the last tag: `git diff <previous-release-tag> docker/Dockerfile`.

## Change Log

**Important**: Any merge into `main` deletes any edits to the draft changelog.
Once you are ready to tag a release, copy the draft changelog into `CHANGELOG.md`.

We follow the [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) format.

We use [the Release Drafter workflow](https://github.com/marketplace/actions/release-drafter) to automatically create a [draft changelog](https://github.com/ZcashFoundation/zebra/releases).

To create the final change log:
- [x] Copy the draft changelog into `CHANGELOG.md`
- [x] Delete any trivial changes. Keep the list of those, to include in the PR
- [x] Combine duplicate changes
- [x] Edit change descriptions so they are consistent, and make sense to non-developers
- [x] Check the category for each change
  - Prefer the "Fix" category if you're not sure

<details>
   
#### Change Categories

From "Keep a Changelog":
* `Added` for new features.
* `Changed` for changes in existing functionality.
* `Deprecated` for soon-to-be removed features.
* `Removed` for now removed features.
* `Fixed` for any bug fixes.
* `Security` in case of vulnerabilities.

</details>

### Create the Release PR

After you have the version increments and the updated changelog:

- [x] Push the version increments and the updated changelog into a branch
      (name suggestion, example: `v1.0.0-alpha.0-release`)
- [x] Create a release PR by adding `&template=release-checklist.md` to the
      comparing url ([Example](https://github.com/ZcashFoundation/zebra/compare/v1.0.0-alpha.0-release?expand=1&template=release-checklist.md)).
  - [x] Add the list of deleted changelog entries as a comment to make reviewing easier.
  - [x] Also add the list of not-bumped crates as a comment (can use the same comment as the previous one).
- [x] Turn on [Merge Freeze](https://www.mergefreeze.com/installations/3676/branches).
- [x] Once the PR is ready to be merged, unfreeze it [here](https://www.mergefreeze.com/installations/3676/branches).
      Do not unfreeze the whole repository.

### Create the Release

- [ ] Once the PR has been merged,
      create a new release using the draft release as a base,
      by clicking the Edit icon in the [draft release](https://github.com/ZcashFoundation/zebra/releases)
- [ ] Set the tag name to the version tag,
      for example: `v1.0.0-alpha.0`
- [ ] Set the release to target the `main` branch
- [ ] Set the release title to `Zebra ` followed by the version tag,
      for example: `Zebra 1.0.0-alpha.0`
- [ ] Copy the final changelog of this release to the release description
      (starting just _after_ the title `## [Zebra ...`)
- [ ] Mark the release as 'pre-release' (until we are no longer alpha/beta)
- [ ] Publish the release

## Final Testing

- [ ] After tagging the release, test that the exact `cargo install` command in `README.md` works
      (`--git` behaves a bit differently to `--path`)
- [ ] Turn off [Merge Freeze](https://www.mergefreeze.com/installations/3676/branches) for the whole repository

If the build fails after tagging:
1. Fix the build
2. Increment versions again, following these instructions from the start
3. Update `README.md` with a **new** git tag
4. Update `CHANGELOG.md` with details about the fix
5. Tag a **new** release

### CODE REVIEW

**Comment by oxarbitrage** [comment]
All crates were bumped.

The following pull requests were left out of the chagelog:

- Make the release checklist shorter and hide some details (#4712)
- Fix walled -> walletd types (#4746)
- build(deps): bump semver from 1.0.10 to 1.0.11 (#4723)
- build(deps): bump reviewdog/action-actionlint from 1.26.0 to 1.27.0 (#4722)
- fix(clippy): resolve or disable new nightly clippy lints (#4739)

File: CHANGELOG.md
Comment (teor2345) [inline CHANGELOG.md]: We might want to say that it's a length limit on a network message:

```suggestion
- Limit the length of the `reject` network command's `message` and `reason` fields (#4687)
```

**Review by teor2345 — CHANGES_REQUESTED** [review]
This release PR looks great!

But looks like syncing still doesn't work reliably:
> 2022-07-09T15:17:43.372416Z WARN{net="Main"}:zebrad::components::sync::progress: chain updates have stalled, state height has not increased for 90 minutes. Hint: check your network connection, and your computer clock and time zone sync_percent=99.928% current_height=Height(1730424) network_upgrade=Nu5 time_since_last_state_block=1h 30m target_block_spacing=PT75S max_block_spacing=None is_syncer_stopped=false

https://github.com/ZcashFoundation/zebra/runs/7264117030?check_suite_focus=true#step:6:311

We're making this release to fix a major bug that stops Zebra syncing. Should we wait until we've fixed all the major syncing bugs before doing the release?

(I think PR #4750 will fix these bugs, I'm hoping I'll get it finished today.)

**Comment by teor2345** [comment]
The syncer hang might also be caused by deadlocks in the `ErrorSlot`, which will be fixed by PR #3263.

We'll know after we merge PR #4750.

**Review by teor2345 — CHANGES_REQUESTED** [review]
We need to add all the changes from 21 July onwards to the CHANGELOG:
https://github.com/ZcashFoundation/zebra/commits/main

File: README.md
Comment (teor2345) [inline README.md]:
```suggestion
- 100 GB+ available disk space for building binaries and storing cached chain state
```

**Comment by teor2345** [comment]
I just noticed that my Zebra instance is using between 35 - 70 GB of disk space, so I updated the disk and network usage in the README and CHANGELOG.

File: README.md
Comment (daira) [reply README.md]: I suggest more than that, given the rate at which chain state size is increasing. Maybe 400 GB+? Or maybe make a distinction between the space that is currently needed and the space that is likely to be needed in future:
```suggestion
- at least 100 GB available disk space for building binaries and storing cached chain state
  (in the longer term, several hundred GB are likely to be needed).
```

File: CHANGELOG.md
Comment (oxarbitrage) [inline CHANGELOG.md:10]: Not true yet.

File: CHANGELOG.md
Comment (teor2345) [inline CHANGELOG.md]: Copying Daira's suggestions here as well:

```suggestion
Zebra now uses around 50 - 100 GB of disk space, because many large transactions were recently added to the block chain. (In the longer term, several hundred GB are likely to be needed.)
```

File: README.md
Comment (teor2345) [inline README.md]:
```suggestion
- Initial sync: 40 GB download (in the longer term, several hundred GB are likely to be downloaded).
```

File: CHANGELOG.md
Comment (teor2345) [reply CHANGELOG.md:10]: I think it's ok to just get CI passing for this release, and do more performance improvements in the next release.

What do you think?

Comment (oxarbitrage) [reply CHANGELOG.md:10]: Yes i am fine with that, if we can fix the CI, then we just need to change the description and add the items that we merged and are not in the changelog yet.

Comment (oxarbitrage) [reply CHANGELOG.md:10]: Also, we should add what it is taking the most time to the known issues.

Comment (teor2345) [reply CHANGELOG.md:10]: Here are some of the known issues and ongoing investigations:
- #4824
- #4822
- #4794
- #4784

RPCs might also be slower than they used to be, we need to check:
- #4831

And here are some of the ongoing investigations:
- #4823
- #4747
- #4583

Comment (oxarbitrage) [reply CHANGELOG.md:10]: I added this list at https://github.com/ZcashFoundation/zebra/pull/4765/commits/71748e6a9572bbb93b563eaf5c7338d27cd904ba

**Comment by oxarbitrage** [comment]
I updated the changelog. In addition to https://github.com/ZcashFoundation/zebra/pull/4765#issuecomment-1179308217 the following tickets are not in the changelog list:

docs(release): Add checkpoint update to the PR release draft template (#4800)
fix(sync): Set the full verify concurrency limit to 5 blocks (#4758)
fix(sync): Temporarily set full verification concurrency to 30 blocks (#4726)
fix(build): Remove duplicate rayon dependency from zebra-chain (#4811)
build(deps): bump tj-actions/changed-files from 23 to 24 (#4807)
build(deps): bump sentry & sentry-tracing from 0.26.0 to 0.27.0 (#4741)
cleanup(difficulty): Remove redundant testnet minimum difficulty implementation (#4760)
Fix up links in some tokio-console and tracing docs (#4748)
add(log): Log the amount of time it takes to rebuild note commitment trees after a chain fork (#4795)
add(log): Log when state requests take a long time (#4815)
fix(test): allow the mempool_transaction_expiration test to get network requeststs in any order (#4830)

I think we need at least https://github.com/ZcashFoundation/zebra/pull/4828 before making the release.

**Comment by teor2345** [comment]
We'll also need #4840, there are two things that aren't quite fixed.

**Review by teor2345 — DISMISSED** [review]
Thanks for all your work on this release, looks great!

**Comment by teor2345** [comment]
@mergifyio update

**Comment by teor2345** [comment]
This PR should be merged automatically by Mergify once all the CI finishes.

(Other PRs are still blocked by Merge Freeze until after we tag the release.)

File: CHANGELOG.md
Comment (oxarbitrage) [inline CHANGELOG.md]:
```suggestion
## [Zebra 1.0.0-beta.13](https://github.com/ZcashFoundation/zebra/releases/tag/v1.0.0-beta.13) - 2022-07-29
```

**Review by conradoplg — APPROVED** [review]

**Comment by teor2345** [comment]
I had an incorrect Depends-On line in the PR, it merged after I fixed it.

**Pull request merged**
